### PR TITLE
Update go version

### PR DESCRIPTION
**Pull request by maxlambrecht**
Updated go version in `.go-version`, `go.mod`, and CI workflows.

### CODE REVIEW

**Review by faisal-memon — APPROVED** [review]
Thanks @maxlambrecht

**Comment by faisal-memon** [comment]
@maxlambrecht Seems like some of the integration tests failed.

File: .github/tests/it/client/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21-alpine AS builder
+FROM golang:1.22.1-alpine AS builder
```
Comment (maxlambrecht) [inline .github/tests/it/client/Dockerfile:1]: The version in the Dockerfiles used for the integration tests should be parametrized, but it's out of scope for this PR.

**Comment by maxlambrecht** [comment]
> @maxlambrecht Seems like some of the integration tests failed.

Fixed.

**Pull request merged**